agb/src/display/tiled/mod.rs

```diff
@@ -126,6 +189,140 @@ impl TileSetting {
     }
 }
 
+pub(self) fn find_screenblock_gap(screenblocks: &Bitarray<1>, gap: usize) -> usize {
+    let mut candidate = 0;
+
+    'outer: while candidate < 16 - gap {
+        let starting_point = candidate;
+        for attempt in starting_point..(starting_point + gap) {
+            if screenblocks.get(attempt) == Some(true) {
+                candidate = attempt + 1;
+                continue 'outer;
+            }
+        }
+
+        return candidate;
+    }
+
+    panic!(
+        "Failed to find screenblock gap of at least {} elements",
+        gap
+    );
+}
+
+pub(self) trait TiledMode {
+    const REGULAR_BACKGROUNDS: usize;
+    const AFFINE_BACKGROUNDS: usize;
+    fn screenblocks(&self) -> &RefCell<Bitarray<1>>;
+    fn regular(&self) -> &RefCell<Bitarray<1>>;
+    fn affine(&self) -> &RefCell<Bitarray<1>>;
+}
+
+pub trait RegularTiledMode {
+    fn regular_background(
+        &self,
+        priority: Priority,
+        size: RegularBackgroundSize,
+    ) -> MapLoan<'_, RegularMap>;
+}
+
+pub trait AffineTiledMode {
+    fn affine_background(
+        &self,
+        priority: Priority,
+        size: AffineBackgroundSize,
+    ) -> MapLoan<'_, AffineMap>;
+}
+
+// withoutboats: https://internals.rust-lang.org/t/const-generics-where-restrictions/12742/6
+pub struct If<const B: bool>;
+pub trait True {}
+impl True for If<true> {}
```

Comment:
that's cool. Does complicate the interface quite a lot though so wondering if the trick is worth it or not :)